test/test_diff.py

```diff
@@ -689,3 +689,115 @@ def test_compare_reports_no_diff():
+    baseline_report = {
+        'git': {
+            'branch': 'master',
+            'head': {
+                'id': 'UNUSED'
+            }
+        },
+        'repo_token': 'UNUSED',
+        'service_job_number': '',
+        'service_name': '',
+        'service_number': '',
+        'source_files': [
+            {
+                'functions': [
+                    {
+                        'name': 'func1',
+                        'start': 36,
+                        'exec': True
+                    },
+                    {
+                        'name': 'func2',
+                        'start': 64,
+                        'exec': False
+                    }
+                ],
+                'branches': [],
+                'name': 'obj-firefox/dist/include/mozilla/dom/DOMRectListBinding.h',
+                'coverage': [
+                    None,
+                    2,
+                    None,
+                    1,
+                    5
+                ],
+                'source_digest': 'another_generated_number_with_literals'
+            }
+        ]
+    }
+    report = {
+        'git': {
+            'branch': 'master',
+            'head': {
+                'id': 'UNUSED'
+            }
+        },
+        'repo_token': 'UNUSED',
+        'service_job_number': '',
+        'service_name': '',
+        'service_number': '',
+        'source_files': [
+            {
+                'functions': [
+                    {
+                        'name': 'func1',
+                        'start': 36,
+                        'exec': True
+                    },
+                    {
+                        'name': 'func2',
+                        'start': 64,
+                        'exec': True
+                    }
+                ],
+                'branches': [],
+                'name': 'obj-firefox/dist/include/mozilla/dom/DOMRectListBinding.h',
+                'coverage': [
+                    None,
+                    2,
+                    None,
+                    5,
+                    6
+                ],
+                'source_digest': 'another_generated_number_with_literals2'
+            }
+        ]
+    }
+    assert diff.compare_reports(baseline_report, report, True) == {
+        'git': {
+            'branch': 'master',
+            'head': {
+                'id': 'UNUSED'
+            }
+        },
+        'repo_token': 'UNUSED',
+        'service_job_number': '',
+        'service_name': '',
+        'service_number': '',
+        'source_files': [
+            {
+                'functions': [
+                    {
+                        'name': 'func2',
+                        'start': 64,
+                        'exec': True
+                    }
+                ],
+                'branches': [],
+                'name': 'obj-firefox/dist/include/mozilla/dom/DOMRectListBinding.h',
+                'coverage': [
+                    None,
```

Comment:
Could you also add a test that returns an array that is not completely filled with None and 0?
E.g. `[ None, 2, None, 1, 0 ]` for baseline and `[ None, 2, None, 5, 6 ]` for report.

Reply:
Oops. It seems that I forgot to set diff as 1 if hit counts > 0.

Reply:
Right! This shows why unit tests are important :smile: